src/ecdsa.rs

```diff
@@ -1,32 +1,50 @@
 use crate::bindings::{
     secp256k1_ec_pubkey_create, secp256k1_ecdsa_sign, secp256k1_ecdsa_signature,
+    secp256k1_ecdsa_signature_parse_compact, secp256k1_ecdsa_signature_serialize_compact,
     secp256k1_ecdsa_verify, secp256k1_pubkey,
 };
 use crate::context::Context;
 use crate::scalar::Scalar;
+use std::array::TryFromSliceError;
 
 #[derive(Debug)]
 /// Errors in ECDSA signature operations
 pub enum Error {
     /// Error occurred due to invalid secret key
     InvalidSecretKey,
+    /// Error occurred during a try from operation
+    TryFrom(String),
+}
+
+impl From<TryFromSliceError> for Error {
+    fn from(e: TryFromSliceError) -> Self {
+        Error::TryFrom(e.to_string())
+    }
 }
 
 /**
 PubKey is a wrapper around libsecp256k1's secp256k1_pubkey struct.
 */
 pub struct PubKey {
+    /// The wrapped secp256k1_pubkey public key
     key: secp256k1_pubkey,
+    // The context associated with the public key
+    context: Context,
```

Comment:
Why store the `Context` for `PubKey`?  We aren't going to use it later for anything, so ephemeral seems best here.